src/pages/beacon-plus.js

```diff
@@ -1,30 +1,84 @@
 import React, { useState } from "react"
 import { useForm } from "react-hook-form"
-import { useDatasets, useFilteringTerms, useBeaconQuery } from "../api/bycon"
+import { useBeaconQuery, useDatasets, useFilteringTerms } from "../api/bycon"
 import Field from "../components/Field"
 import DataFetchSelect from "../components/DataFetchSelect"
 import BiosamplesDataTable from "../components/BiosamplesDataTable"
 import CnvHistogram from "../components/CnvHistogram"
 import Nav from "../components/Nav"
+import requestTypesConfig from "../../config/beacon-plus/requestTypes.yaml"
+import examplesConfig from "../../config/beacon-plus/examples.yaml"
+import { markdownToReact } from "../md"
 
 export default function BeaconPlus() {
-  const { register, handleSubmit, errors, getValues } = useForm()
-  const resultType = getValues("resultType")
+  const { register, handleSubmit, errors, getValues, reset } = useForm()
   const [query, setQuery] = useState(null) // actual valid query
+  // could be the example of the requestType as they have the same shape and effect
+  const [requestConfig, setRequestConfig] = useState(null)
   const { data, error, mutate } = useBeaconQuery(query)
+
+  const resultType = getValues("resultType")
   const isLoading = !data && !error && !!query
   const onSubmit = (formValues) => {
     mutate(null) // mutate and clear current results
-    const { datasetIds, assemblyId, referenceName, filters } = formValues
-    setQuery({ datasetIds, assemblyId, referenceName, filters })
+    setQuery(formValues)
+  }
+
+  const setRequestType = (requestType) => {
+    reset()
+    setRequestConfig(requestType)
+  }
+  const setExample = (example) => {
+    const newParams = Object.fromEntries(
+      Object.entries(example.parameters).map(([k, v]) => [k, v.examplevalue])
+    )
+    reset(newParams)
+    setRequestConfig(example)
   }
 
   return (
     <>
       <Nav />
       <section className="section">
         <div className="container">
+          <article className="message is-dark">
+            <div className="message-body">
+              {requestConfig?.description ? (
+                markdownToReact(requestConfig?.description)
+              ) : (
+                <span>
```

Comment:
Is this "standard" to have defaults in the .js?

Reply:
You mean this syntax? `?.`
This is part of the current spec yes, but pretty new. Will be translated by babel for older browsers.